src/store/inmem/block.zig

```diff
@@ -92,8 +91,78 @@ pub const Block = struct {
     }
 
     fn put(self: *Block, allocator: std.mem.Allocator, lines: []*const Line) !void {
-        // TODO: implement fast path when lines have same fields
+        // If len is zero, nothing to do.
+        if (lines.len == 0) {
+            return;
+        }
+
+        // Fast path if all lines have the same fields
+        if (Block.areSameFields(lines)) {
+            // Extract timestamps
+            for (lines, 0..) |line, i| {
+                self.timestamps[i] = line.timestampNs;
+            }
+
+            // All lines have same fields, process each field column
+            const firstLine = lines[0];
+            var columns = try allocator.alloc(Column, firstLine.fields.len);
+            errdefer allocator.free(columns);
+
+            @memset(columns, .{ .key = "", .values = &[_][]u8{} });
+            errdefer {
+                for (columns) |col| {
+                    if (col.values.len != 0) {
+                        allocator.free(col.values);
+                    }
+                }
+            }
+
+            // First pass: identify which columns are celled
+            var celledMask = try allocator.alloc(bool, firstLine.fields.len);
+            defer allocator.free(celledMask);
 
+            var celledCount: usize = 0;
+            for (firstLine.fields, 0..) |_, fieldIdx| {
+                if (Block.canBeSavedAsCelled(lines, fieldIdx)) {
+                    celledMask[fieldIdx] = true;
+                    celledCount += 1;
+                } else {
+                    celledMask[fieldIdx] = false;
+                }
+            }
+
+            // Second pass: populate columns with regular columns first, then celled
+            var regularIdx: usize = 0;
+            var celledIdx: usize = firstLine.fields.len - celledCount;
+
+            for (firstLine.fields, 0..) |field, fieldIdx| {
+                const isFieldCelled = celledMask[fieldIdx];
```

Comment:
not a field celled, but a column.
like a column is
```
 _
| |
| |
| |
```
but a cell is 
```
[]
```

Reply:
look at a google/excel sheet

```
X key1 (A) key2 (B)
1  42      same        
2  43      same
3  44      same
```

logs lines are 1, 2, 3.
every line had fields key1 and key2.
we built columns from them, then we can shrink key2 to a single cell [key2: same], therefore we can convert the column {key2: same, same, same} to a field having only key value, we don't do it for the same of memory savings. 

so that field is a field everywhere, but a column can become a cell or celledColumn